tests/run-shell.mjs

```diff
@@ -0,0 +1,189 @@
+
+const SHELL_NAME = (function() {
+  switch (JS_SHELL) {
+      case "javascriptcore":
+      case "jsc": {
+          return "javascriptcore";
+    }
+      case "spidermonkey": {
+          return "spidermonkey";
+      }
+      case "v8": {
+          return "v8";
+      }
+      default: {
+          printHelp(`Invalid shell "${JS_SHELL}", choices are: "jsc", "spidermonkey" and "v8)`);
+      }
+  }
+})();
+
+const FILE_PATH = fileURLToPath(import.meta.url);
+const SRC_DIR = path.dirname(path.dirname(FILE_PATH));
+const CLI_PATH = path.join(SRC_DIR, "cli.js");
+
+const BASE_CLI_ARGS_WITH_OPTIONS = [CLI_PATH];
+if (SHELL_NAME != "spidermonkey")
+  BASE_CLI_ARGS_WITH_OPTIONS.push("--");
+Object.freeze(BASE_CLI_ARGS_WITH_OPTIONS);
+
+const GITHUB_ACTIONS_OUTPUT = "GITHUB_ACTIONS_OUTPUT" in process.env;
+
+function log(...args) {
+  const text = args.join(" ")
+  if (GITHUB_ACTIONS_OUTPUT)
+    core.info(styleText("yellow", text))
+  else
+    console.log(styleText("yellow", text))
+}
+
+function logError(...args) {
+  const text = args.join(" ")
+  if (GITHUB_ACTIONS_OUTPUT)
+    core.error(styleText("red", text))
+  else
+    console.error(styleText("red", text))
+}
+
+function logGroup(name, body) {
+  if (GITHUB_ACTIONS_OUTPUT) {
+    core.startGroup(name);
+  } else {
+    log("=".repeat(80))
+    log(name);
+    log(".".repeat(80))
+  }
+  try {
+    return body();
+  } finally {
+    if (GITHUB_ACTIONS_OUTPUT)
+      core.endGroup();
+  } 
+}
+
+const SPAWN_OPTIONS =  { 
+  stdio: ["inherit", "inherit", "inherit"]
+};
+
+function sh(binary, args) {
+  const cmd = `${binary} ${args.join(" ")}`;
+  if (GITHUB_ACTIONS_OUTPUT) {
+    core.startGroup(binary);
+    core.notice(styleText("blue", cmd));
+  } else {
+    console.log(styleText("blue", cmd));
+  }
+  try {
+    const result = spawnSync(binary, args, SPAWN_OPTIONS);
+    if (result.status || result.error) {
+      logError(result.error);
+      throw new Error(`Shell CMD failed: ${binary} ${args.join(" ")}`);
+    }
+  } finally {
+    if (GITHUB_ACTIONS_OUTPUT)
+      core.endGroup()
+  }
+}
+
+async function runTests() {
+    const shellBinary = logGroup(`Installing JavaScript Shell: ${SHELL_NAME}`, testSetup);
+    let success = true;
+    success &&= runTest("Run Complete Suite", () => sh(shellBinary, [CLI_PATH]));
+    success &&= runTest("Run Single Suite", () => {
+      const singleTestArgs = [...BASE_CLI_ARGS_WITH_OPTIONS, "proxy-mobx"];
+      sh(shellBinary, singleTestArgs);
+    });
+    if (!success) {
+      process.exit(1)
+    }
+}
+
+function jsvuOSName() {
+  const osName = () => {
+      switch (os.platform()) {
+          case "win32": return "win";
+          case "darwin": return "mac";
+          case "linux": return "linux";
+          default: throw new Error("Unsupported OS");
+      }
+  };
+  const osArch = () => {
+      switch (os.arch()) {
+          case "x64": return "64";
+          case "ia32": return "32";
```

Comment:
Huh, do other JS engines still support x86 32-bit? JSC doesn't anymore at least. Regardless, I assume this isn't actually run since it wouldn't work for JSC.

Reply:
v8 definitely does... going to remove that since it might be just more confusing than necessary.

Reply:
SM also supports it, FWIW.